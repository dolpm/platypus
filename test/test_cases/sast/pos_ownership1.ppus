pipe test [x: int] |> unit {
  |> ();
}

pipe main [] |> unit {
  int x = 4;
  int z = 4;
   {
     int y = x;

    while (y < 4) {
      {
        int w = 8;
        {
          int a = w;
          test <| [a];
        }
      }
    }
  
   }
   |> ();
}