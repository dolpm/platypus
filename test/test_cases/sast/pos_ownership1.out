pipe test |> [] |> [x: int] |> unit {
  |> (unit : ());
}

pipe main |> [] |> [] |> unit {
  int x <| (int : 4);
  int z <| (int : 4);
{
  int y <| (int : x);
  while (bool : (int : y) < (int : 4))
  {
  {
    int w <| (int : 8);
    (unit : test <| [(int : w)]);
  }
  }
}
  |> (unit : ());
}
